unlock-app/src/services/walletService.js

```diff
@@ -91,6 +94,35 @@ export default class WalletService extends EventEmitter {
     }
   }
 
+  /**
+   * Poll to see if account has changed
+   */
+  async pollForAccountChange(timeout, isServer) {
+    if (isServer) return
+    while (true) {
+      try {
+        await timeout(POLLING_INTERVAL)
+      } catch (e) {
+        return // if the timeout throws, we don't start the loop, this is primarily for testing
+      }
+      try {
+        this.account = await this.checkForAccountChange(this.account)
+      } catch (e) {
+        continue // if the account poll fails, silently ignore it
+      }
+    }
+  }
+
+  async checkForAccountChange(priorAddress) {
+    const accounts = await this.web3.eth.getAccounts()
```

Comment:
I think there is some logic duplicated here with `getAccount`. Any way to DRY this up?

Reply:
[that probably means splitting up `getAccount` and/or `getOrCreateAccount` which is fine.

Reply:
sure I'll see what I can wrangle up